### PR TITLE
Add a merge unit.

### PR DESCRIPTION
**Pull request by partim**
This PR adds a new `merge` unit that merges the payload from all its healthy sources.

### CODE REVIEW

File: src/units/combine.rs [hunk 1]
```diff
@@ -195,6 +195,58 @@ impl metrics::Source for AnyMetrics {
...
+            gate.update(UnitUpdate::Gone).await;
+            return Err(Terminated)
+        }
+        let metrics = gate.metrics();
+        component.register_metrics(metrics.clone());
+
+        loop {
+            {
+                let res = select(
+                    select_all(
```
Comment (DRiKE) [inline src/units/combine.rs:221]: This will wait for _any_ out of _all_ the sources, but it will not wait for all of them. Intuitively, I'd say you actually want the latter? (So, something more like a `join_all` perhaps?)

File: src/payload.rs [hunk 2]
```diff
@@ -1272,13 +1380,37 @@ mod test {
     use super::*;
     use super::testrig::*;
 
+    #[test]
+    fn set_merge() {
```
Comment (DRiKE) [inline src/payload.rs:1384]: Maybe add tests for merging sets of which one and/or both are empty?

File: src/payload.rs [hunk 3]
```diff
@@ -1243,9 +1351,9 @@ pub(crate) mod testrig {
     }
 
     /// Creates an update from a bunch of integers
```
[the remark this answers is not in the document]
Comment (partim) [reply src/payload.rs:1353]: Changed it anyway …

File: src/units/combine.rs (hunk 1 above)
Comment (partim) [reply src/units/combine.rs:221]: Shouldn’t it update the merged set every time any of the units updates?

**Review by DRiKE — APPROVED** [review]
